src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1191,7 +1188,16 @@ fn calculate_run_custom_build<'a, 'cfg>(
 fn build_script_local_fingerprints<'a, 'cfg>(
     cx: &mut Context<'a, 'cfg>,
     unit: &Unit<'a>,
-) -> (Box<dyn MyFnOnce + Send>, bool) {
+) -> (
+    Box<
+        dyn FnOnce(
+                &BuildDeps,
+                Option<&dyn Fn() -> CargoResult<String>>,
+            ) -> CargoResult<Option<Vec<LocalFingerprint>>>
+            + Send,
+    >,
```

Comment:
This is rustfmt's work -- I ... well, I can't say I *like* it, but I also don't know that it can be made much better. This is a pretty ugly type.